Foundation/FileManager+Win32.swift

```diff
@@ -286,12 +292,20 @@ extension FileManager {
     internal func _canonicalizedPath(toFileAtPath path: String) throws -> String {
         var hFile: HANDLE = INVALID_HANDLE_VALUE
         path.withCString(encodedAs: UTF16.self) { link in
-            hFile = CreateFileW(link, GENERIC_READ, DWORD(FILE_SHARE_WRITE), nil, DWORD(OPEN_EXISTING), DWORD(FILE_FLAG_BACKUP_SEMANTICS), nil)
+            hFile = CreateFileW(link,
+                                0,
+                                DWORD(FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE),
+                                nil,
+                                DWORD(OPEN_EXISTING),
+                                // BACKUP_SEMANTICS are (confusingly) required
+                                // in order to receive a handle to a directory
```

Comment:
I would've put the comment before the `CreateFileW` call.  BTW, Foundation doesn't do the one parameter per line for the arguments.